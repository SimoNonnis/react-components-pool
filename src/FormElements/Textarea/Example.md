Textarea with no Label

```jsx
<Textarea id="textarea" label="Notes" />
```

Textarea with required and with error

```jsx
<Textarea id="textarea1" label="Notes" error="Error" required />
```

Textarea with disabled

```jsx
<Textarea id="textarea2" label="Notes" required disabled />
```
