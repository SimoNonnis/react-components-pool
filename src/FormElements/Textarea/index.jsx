import React from 'react';
import PropTypes from 'prop-types';

import Label from '../Label';
import InputFeedback from '../InputFeedback';

import { TextareaStyled } from '../_common/style';

const propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  className: PropTypes.string,
  label: PropTypes.string,
  error: PropTypes.string,
  rows: PropTypes.number,
  disabled: PropTypes.bool,
  required: PropTypes.bool,
};

const defaultProps = {
  className: null,
  name: null,
  label: null,
  error: null,
  rows: 5,
  disabled: false,
  required: false,
};

const Textarea = ({ className, id, label, error, name, rows, disabled, required, ...props }) => (
  <div className={className}>
    <Label htmlFor={id} error={!!error} disabled={disabled} required={required}>
      {label}
    </Label>
    <TextareaStyled
      id={id}
      name={name}
      error={!!error}
      autoComplete="nope"
      rows={rows}
      disabled={disabled}
      {...props}
    />
    <InputFeedback error={error} />
  </div>
);

Textarea.propTypes = propTypes;
Textarea.defaultProps = defaultProps;

export default Textarea;
