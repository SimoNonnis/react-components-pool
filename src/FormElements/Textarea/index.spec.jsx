/* eslint-env jest */
import React from 'react';

import 'jest-styled-components';

import Textarea from './';

describe('Test <Textarea />', () => {
  it('should render Textarea correctly with only required props', () => {
    const wrapper = shallow(<Textarea id="testcomment" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Textarea correctly with a label', () => {
    const wrapper = shallow(<Textarea id="testcomment" label="Comment" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Textarea correctly if disabled', () => {
    const wrapper = shallow(<Textarea id="testcomment" label="Comment" disabled />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Textarea correctly with a label and required flag', () => {
    const wrapper = shallow(<Textarea id="testcomment" label="Comment" required />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Textarea correctly with a label and an error', () => {
    const wrapper = shallow(<Textarea id="testcomment" label="Comment" error="New error" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Textarea correctly passing rows value', () => {
    const wrapper = shallow(<Textarea id="testcomment" label="Comment" rows={10} />);
    expect(wrapper).toMatchSnapshot();
  });
});
