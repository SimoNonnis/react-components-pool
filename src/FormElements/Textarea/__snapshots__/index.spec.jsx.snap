// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Textarea /> should render Textarea correctly if disabled 1`] = `
<div
  className={null}
>
  <Label
    className={null}
    disabled={true}
    error={false}
    htmlFor="testcomment"
    required={false}
  >
    Comment
  </Label>
  <styled.textarea
    autoComplete="nope"
    disabled={true}
    error={false}
    id="testcomment"
    name={null}
    rows={5}
  />
  <InputFeedback
    className={null}
    error={null}
  />
</div>
`;

exports[`Test <Textarea /> should render Textarea correctly passing rows value 1`] = `
<div
  className={null}
>
  <Label
    className={null}
    disabled={false}
    error={false}
    htmlFor="testcomment"
    required={false}
  >
    Comment
  </Label>
  <styled.textarea
    autoComplete="nope"
    disabled={false}
    error={false}
    id="testcomment"
    name={null}
    rows={10}
  />
  <InputFeedback
    className={null}
    error={null}
  />
</div>
`;

exports[`Test <Textarea /> should render Textarea correctly with a label 1`] = `
<div
  className={null}
>
  <Label
    className={null}
    disabled={false}
    error={false}
    htmlFor="testcomment"
    required={false}
  >
    Comment
  </Label>
  <styled.textarea
    autoComplete="nope"
    disabled={false}
    error={false}
    id="testcomment"
    name={null}
    rows={5}
  />
  <InputFeedback
    className={null}
    error={null}
  />
</div>
`;

exports[`Test <Textarea /> should render Textarea correctly with a label and an error 1`] = `
<div
  className={null}
>
  <Label
    className={null}
    disabled={false}
    error={true}
    htmlFor="testcomment"
    required={false}
  >
    Comment
  </Label>
  <styled.textarea
    autoComplete="nope"
    disabled={false}
    error={true}
    id="testcomment"
    name={null}
    rows={5}
  />
  <InputFeedback
    className={null}
    error="New error"
  />
</div>
`;

exports[`Test <Textarea /> should render Textarea correctly with a label and required flag 1`] = `
<div
  className={null}
>
  <Label
    className={null}
    disabled={false}
    error={false}
    htmlFor="testcomment"
    required={true}
  >
    Comment
  </Label>
  <styled.textarea
    autoComplete="nope"
    disabled={false}
    error={false}
    id="testcomment"
    name={null}
    rows={5}
  />
  <InputFeedback
    className={null}
    error={null}
  />
</div>
`;

exports[`Test <Textarea /> should render Textarea correctly with only required props 1`] = `
<div
  className={null}
>
  <Label
    className={null}
    disabled={false}
    error={false}
    htmlFor="testcomment"
    required={false}
  />
  <styled.textarea
    autoComplete="nope"
    disabled={false}
    error={false}
    id="testcomment"
    name={null}
    rows={5}
  />
  <InputFeedback
    className={null}
    error={null}
  />
</div>
`;
